### PR TITLE
GEODE-8538: Create test to validate ordering of redis pipeline commands

### PR DESCRIPTION
**Pull request by jhutchison**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/CommandPipeliningIntegrationTest.java [hunk 1]
```diff
@@ -93,6 +98,42 @@ public void whenPipelining_commandResponsesAreNotCorrupted() {
...
+
+      verifyResultOrder(NUMBER_OF_COMMANDS_IN_PIPELINE, results);
+      numberOfPipeLineRequests--;
+    } while (numberOfPipeLineRequests > 0);
+
+    jedis.flushAll();
+    jedis.close();
+  }
+
+  private void verifyResultOrder(int NUMBER_OF_COMMAND_IN_PIPELINE, List<Object> results) {
```
Comment (ringles) [inline geode-redis/src/integrationTest/java/org/apache/geode/redis/CommandPipeliningIntegrationTest.java:128]: NUMBER_OF_COMMAND_IN_PIPELINE looks like a constant, but isn't. It should be something like "numberOfCommands" or "numberOfCommandsInPipeline".

**Review by ringles — CHANGES_REQUESTED** [review]
One little tweak, otherwise looks good.

File: geode-redis/src/integrationTest/java/org/apache/geode/redis/CommandPipeliningIntegrationTest.java (hunk 1 above)
Comment (jhutchison) [reply geode-redis/src/integrationTest/java/org/apache/geode/redis/CommandPipeliningIntegrationTest.java:128]: made final.  Thanks!

**Comment by jdeppe-pivotal** [comment]
I think the changes need to be rebased onto develop to avoid some of the strange junit failures.

**Review by dschneider-pivotal — APPROVED** [review]

**Pull request merged**